sdk/openai/Azure.AI.OpenAI/src/Custom/AzureCognitiveSearchChatExtensionConfiguration.cs

```diff
@@ -0,0 +1,108 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+// <auto-generated/>
+
+#nullable disable
+
+using System;
+using Azure.Core;
+
+namespace Azure.AI.OpenAI
+{
+    /// <summary>
+    /// A specific representation of configurable options for Azure Cognitive Search when using it as an Azure OpenAI chat
+    /// extension.
+    /// </summary>
+    public partial class AzureCognitiveSearchChatExtensionConfiguration : AzureChatExtensionConfiguration
```

Comment:
I see that all the `AzureCognitiveSearch*` classes are custom code. Is the only reason for this because the emitter thinks that the models aren't used and therefore won't auto-generate the classes for them? If so, we should consider defining them in the TypeSpec as polymorphic. Ultimately, it sounds like this is truly how it should be, and it would have the side effect of auto-generating all these classes.

Reply:
(Copy/pasting a few times) per the great offline discussion, let's defer further refinement of this for fast followup after we get things out the door.

(For this one, we'll need upstream service definitions to have tighter constraints for us to reflect nice, tidy relationships -- hopefully we can do that, but for now we can leave it as seemingly independent helper code)